apps/ejabberd/src/foreign_event/mod_foreign_http.erl

```diff
@@ -28,9 +28,11 @@
 %%--------------------------------------------------------------------
 
 make_request(Request, OnResponse) ->
-    {Host, Path, Method, Headers, Body} = parse(Request),
-    Response = do_make_request(Host, Path, Method, Headers, Body),
-    respond(encode(Response), OnResponse),
+    spawn(fun() ->
```

Comment:
If you need asynchronous request, there is one thing I need to point out: You really shouldn't do that using `spawn` directly. This way you just cannot control the number of process in the system what may be dangerous or at least "not fair" to other modules. There should be a pool for that.